### DISCUSSION TITLE
How do I tell the Vector MQTT sink to use protocol 3.1.1

### DISCUSSION THREAD

**Discussion by mrbluecoat** [vectordotdev/vector#21378] [Q&A]
I believe the Vector MQTT sink is using protocol 5. How can I configure it to use 3.1.1 instead (aka, [confusingly,](https://www.hivemq.com/blog/the-history-of-mqtt-part-3-mqtt5-next-generation-mqtt/) "V4")?

Would I replace https://github.com/vectordotdev/vector/blob/438bc8b47c294a954fbb8bbb56f573e74ee749d0/src/sinks/mqtt/config.rs#L4 

`use rumqttc::{` with `use rumqttc::{mqttbytes::v4::`

like https://github.com/ErikWegner/homecontrol-ui/blob/f76f6ff26974b0c52678ba9c312e359cfed9db1b/server/src/mqtta/actor.rs#L6

**Comment by jszwedko** [comment]
I think this would be a feature request. Would you want to open one? See https://github.com/vectordotdev/vector/issues/new?assignees=&labels=type%3A+feature&projects=&template=feature.yml

**Comment by mrbluecoat** [comment]
All set: https://github.com/vectordotdev/vector/issues/21395
